### PR TITLE
example: Replace grommet with sx styled components

### PR DESCRIPTION
**Pull request by tbergq**
- Currently deactivate client side routing, since it doesn't work with sx
- Next up: remove gromment-icons, so we can remove styled-components

### CODE REVIEW

File: src/example-relay/src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  iconRight,
+  type = 'button',
+  ...rest
+}: Props): React.Node {
+  return (
+    <button
+      // eslint-disable-next-line react/button-has-type
+      type={type}
+      data-testid={dataTest}
+      className={styles('button', 'primary', rest.disabled ? 'disabled' : 'button')} // TODO: Allow conditional style in sx
```
Comment (mrtnzlml) [inline src/example-relay/src/components/Button.js:29]: This should be already possible in the latest SX version. Could you please update this code? Or is there still some problem we need to fix?

Comment (tbergq) [reply src/example-relay/src/components/Button.js:29]: No, I forgot to fix it after rebasing, thanks for pointing it out 😊

File: src/example-relay/src/pages/_app.js [hunk 2]
```diff
@@ -102,9 +71,21 @@ const styles = sx.create({
...
     maxWidth: '100%',
-    padding: 12,
+    padding: 'var(--space-small)',
   },
   boxWarning: {
     backgroundColor: '#FFCA28',
   },
+  padMedium: {
+    padding: 'var(--space-small)',
+    [tablet]: {
```
Comment (mrtnzlml) [inline src/example-relay/src/pages/_app.js:81]: This is an interesting technique. I kinda like it. :)

File: src/example-relay/src/components/Button.js (hunk 1 above)
Comment (tbergq) [reply src/example-relay/src/components/Button.js]: 🚀

File: src/example-relay/src/pages/_app.js (hunk 2 above)
Comment (tbergq) [reply src/example-relay/src/pages/_app.js:81]: Yeah, it seems like the best way to keep your breakpoints in sync throughout the app 🙂

File: src/example-relay/src/components/TextInput.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <label>
+      <Text size="small">{label}</Text>
+      <input {...rest} className={styles('input')} onChange={handleChange} />
+    </label>
+  );
+}
+
+const styles = sx.create({
+  label: {
```
Comment (mrtnzlml) [inline src/example-relay/src/components/TextInput.js:29]: Is this style unused? 🤔 

We need to figure out some simple lint.

Comment (tbergq) [reply src/example-relay/src/components/TextInput.js:29]: Yes, it is, some linting could definitely be added at some point yes 👍

Comment (tbergq) [reply src/example-relay/src/components/TextInput.js]: 🚀

**Review by mrtnzlml — APPROVED** [review]
It's an interesting overview of SX usage. It will definitely help us to develop it further. Good job!

**Pull request merged**